src/main/java/core/basesyntax/model/Operation.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.model;
+
+public enum Operation {
+    BALANCE("b"),
+
+    SUPPLY("s"),
+
+    PURCHASE("p"),
+
+    RETURN("r");
+
+    private String code;
+
+    Operation(String code) {
+        this.code = code;
+    }
+
+    public static Operation getOperationFromCode(String operationValue) {
+        return switch (operationValue) {
+            case "b" -> BALANCE;
+            case "s" -> SUPPLY;
+            case "p" -> PURCHASE;
+            case "r" -> RETURN;
+            default -> throw new RuntimeException("You have entered a non-existent transaction."
+                    + operationValue
+                    + "Available values: BALANCE, SUPPLY, PURCHASE, RETURN");
+        };
```

Comment:
not the best solution here
iterate with for-each, throw illegal argument exception 
with current approach you would need to modify this method everytime operation set changes